x/blob/types/builder.go

```diff
@@ -202,6 +202,12 @@ func (k *KeyringSigner) SetKeyringAccName(name string) {
 	k.keyringAccName = name
 }
 
+// SetEncodingConfig manually overides the encoding config of the underlying
+// signer
+func (k *KeyringSigner) SetEncodingConfig(encCfg encoding.Config) {
+	k.encCfg = encCfg
+}
```

Comment:
kinda hacky. the entire keyring signer can be refactored or removed entirely now that we don't have to malleate txs, as its simply a script to programatically sign things. We still use it a lot in tests, and I think celestia-node uses it when creating a PFB, but we could probably replace those with a simple function instead.

https://github.com/celestiaorg/celestia-app/issues/1162